WordPress/gutenberg
# Use state for dimensions visualizers

## Description


Fixes #31839 by using component state instead of block attributes for storing the visualizer information.

Props added are undocumented and marked as `__experimental` because the box visualizer is also marked as `__experimental`. I expect that when the visualizer API is finalized, we can finalize these props and document them as something that the hook provides for components that wish to use them.

## How has this been tested?




1. Add cover block
2. Add padding
3. See that attributes are not updated when hovering the padding controls, but UI still shows.
4. The "Save draft" button shouldn't be affected and an undo state shouldn't be added either since attributes aren't updated.

## Screenshots 

https://user-images.githubusercontent.com/5129775/126219283-5a2d4d71-0a0b-4962-8bdd-7fc3778cecb4.mp4

## Types of changes





Bug fix

## Checklist:
